packages/pipeline-editor/src/PipelineSubmissionDialog.tsx

```diff
@@ -14,65 +14,44 @@
  * limitations under the License.
  */
 
-import { Dialog } from '@jupyterlab/apputils';
-import { Widget, PanelLayout } from '@lumino/widgets';
-
-/**
- * Pipeline submission dialog widget
- */
-export class PipelineSubmissionDialog extends Widget
-  implements Dialog.IBodyWidget<any> {
-  constructor(props: any) {
-    super(props);
-
-    const layout = (this.layout = new PanelLayout());
-    const htmlContent = this.getHtml(props);
-    // Set default runtime to kfp, since list is dynamically generated
-    (htmlContent.getElementsByClassName(
-      'elyra-form-runtime-config'
-    )[0] as HTMLSelectElement).value = 'kfp';
-
-    layout.addWidget(new Widget({ node: htmlContent }));
-  }
-
-  getValue(): any {
-    return {
-      pipeline_name: (document.getElementById(
-        'pipeline_name'
-      ) as HTMLInputElement).value,
-      runtime_config: (document.getElementById(
-        'runtime_config'
-      ) as HTMLInputElement).value
-    };
-  }
-
-  getHtml(props: any): HTMLElement {
-    const htmlContent = document.createElement('div');
-    const br = '<br/>';
-    let runtime_options = '';
-    const runtimes = props['runtimes'];
-
-    for (const key in runtimes) {
-      runtime_options =
-        runtime_options +
-        `<option value="${runtimes[key]['name']}">${runtimes[key]['display_name']}</option>`;
-    }
-
-    const content =
-      '' +
-      '<label for="pipeline_name">Pipeline Name:</label>' +
-      br +
-      '<input type="text" id="pipeline_name" name="pipeline_name" placeholder="Pipeline Name" data-form-required/>' +
-      br +
-      br +
-      '<label for="runtime_config">Runtime Config:</label>' +
-      br +
-      '<select id="runtime_config" name="runtime_config" class="elyra-form-runtime-config" data-form-required>' +
-      runtime_options +
-      '</select>';
-
-    htmlContent.innerHTML = content;
-
-    return htmlContent;
-  }
-}
+import * as React from 'react';
+
+import { IRuntime } from './PipelineEditorWidget';
+
+type Props = {
+  runtimes: IRuntime[];
+};
+
+const PipelineSubmissionDialog = ({ runtimes }: Props): JSX.Element => {
```

Comment:
Given you've made all the dialogs `const` of `JSX.Element` instead of classes should we keep all the dialogs in a `dialogs.tsx` file instead of separate files? Especially since `PipelineSubmissionDialog.tsx` and `PipelineExportDialog.tsx` were named after the class they contained.

Reply:
Do you prefer classes? I can switch to classes or put all the dialogs in one file or both or neither haha, up to you 😄

I normally use function components instead of class components and I usually keep one component per file

I also use normal functions instead of arrow functions, but eslint didn't like that:
```tsx
function PipelineSubmissionDialog(props) {
  ...
}
```

Reply:
That makes sense, I personally don't have a preference, but I believe we and JupyterLab usually use classes so I think we should be consistent (unless making this a class would bloat the code, in which case lets keep it as is)

Reply:
I would not recommend stashing all dialogs on the same file, I believe what I have seen on other extensions is a "component" per file, so a dialog or widget per file.